samples/cellular/nrf_cloud_mqtt_multi_service/src/fota_support_coap.c

```diff
@@ -0,0 +1,527 @@
+		img_type = DFU_TARGET_IMAGE_TYPE_MODEM_DELTA;
+		break;
+	case NRF_CLOUD_FOTA_MODEM_FULL:
+		img_type = DFU_TARGET_IMAGE_TYPE_FULL_MODEM;
+		break;
+	default:
+		LOG_ERR("Unhandled FOTA type: %d", job.type);
+		return -EFTYPE;
+	}
+
+	LOG_INF("Starting FOTA download of %s/%s", job.host, job.path);
+	int err = fota_download_start_with_image_type(job.host, job.path,
+		CONFIG_NRF_CLOUD_SEC_TAG, 0, FOTA_DL_FRAGMENT_SZ,
+		img_type);
+
+	if (err != 0) {
+		LOG_ERR("Failed to start FOTA download, error: %d", err);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int wait_for_download(void)
+{
+	LOG_INF("Waiting for FOTA download to complete");
+	int err = k_sem_take(&fota_download_sem,
+			K_MINUTES(CONFIG_COAP_FOTA_DL_TIMEOUT_MIN));
+	if (err == -EAGAIN) {
+		fota_download_cancel();
+		return -ETIMEDOUT;
+	} else if (err != 0) {
+		LOG_ERR("k_sem_take error: %d", err);
+		return -ENOLCK;
+	}
+
+	return 0;
+}
+
+static void handle_download_succeeded_and_reboot(void)
+{
+	int err;
+
+	/* Save the pending FOTA job info to flash */
+	memcpy(pending_job.id, job.id, NRF_CLOUD_FOTA_JOB_ID_SIZE);
+	pending_job.type = job.type;
+	pending_job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	pending_job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_CLEAR;
+
+	err = nrf_cloud_bootloader_fota_slot_set(&pending_job);
+	if (err) {
+		LOG_WRN("Failed to set B1 slot flag, BOOT FOTA validation may be incorrect");
+	}
+
+	(void)lte_lc_deinit();
+
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+	if (job.type == NRF_CLOUD_FOTA_MODEM_FULL) {
+		LOG_INF("Applying full modem FOTA update...");
+		err = nrf_cloud_fota_fmfu_apply();
+		if (err) {
+			LOG_ERR("Failed to apply full modem FOTA update %d", err);
+			pending_job.validate = NRF_CLOUD_FOTA_VALIDATE_FAIL;
+		} else {
+			pending_job.validate = NRF_CLOUD_FOTA_VALIDATE_PASS;
+		}
+	}
+#endif
+
+	err = save_pending_job();
+	if (err) {
+		LOG_WRN("FOTA job will be marked as successful without validation");
+		fota_status_details = FOTA_STATUS_DETAILS_NO_VALIDATE;
+		(void)update_job_status();
+	}
+
+	LOG_INF("Rebooting in %us to complete FOTA update...", FOTA_REBOOT_S);
+	LOG_PANIC();
+	k_sleep(K_SECONDS(FOTA_REBOOT_S));
+	sys_reboot(SYS_REBOOT_COLD);
+}
+
+static void cleanup(void)
+{
+	nrf_cloud_coap_fota_job_free(&job);
+}
+
+static void wait_after_job_update(void)
+{
+	/* Job operations can take up to 30s to be processed */
+	LOG_INF("Waiting %u seconds for job update to be processed by nRF Cloud...", JOB_WAIT_S);
+	cleanup();
+	k_sleep(K_SECONDS(JOB_WAIT_S));
+}
+
+static void error_reboot(void)
+{
+	LOG_INF("Rebooting due to error in %us...", ERROR_REBOOT_S);
+	(void)lte_lc_deinit();
+	LOG_PANIC();
+	k_sleep(K_SECONDS(ERROR_REBOOT_S));
+	sys_reboot(SYS_REBOOT_COLD);
+}
+
+int coap_fota_process(void)
+{
+	int err;
+
+	/* If a FOTA job is in progress, handle it first */
+	if (validate_in_progress_job()) {
+		err = update_job_status();
+		if (err) {
+			LOG_ERR("Error; must reboot: %d", err);
```

Comment:
`LOG_ERR("Error updating FOTA job status: %d", err);`

the reboot is mentioned in the `error_reboot` log msg.